File: backend/src/main/java/com/example/swp/features/audit_log/AuditLogService.java
package com.example.swp.features.audit_log;

import com.example.swp.features.audit_log.dto.request.CreateAuditLogRequest;
import com.example.swp.features.audit_log.dto.response.AuditLogResponse;

import java.util.List;

public interface AuditLogService {
    AuditLogResponse createAuditLog(CreateAuditLogRequest request);
    List<AuditLogResponse> getAllAuditLogs();
    List<AuditLogResponse> getAuditLogsByUser(Long userId);
    org.springframework.data.domain.Page<AuditLogResponse> getAuditLogsByEvent(Long eventId, org.springframework.data.domain.Pageable pageable);
    void logAction(String action, String entityType, Long entityId, String oldValue, String newValue, Long eventId);
    org.springframework.web.servlet.mvc.method.annotation.SseEmitter subscribeToEventLogs(Long eventId);
    
    default void logAction(String action, String entityType, Long entityId, String oldValue, String newValue) {
        logAction(action, entityType, entityId, oldValue, newValue, null);
    }
}

File: frontend/src/services/AuditLogService.ts
import api from './api';

const API_URL = '/audit-logs';

export interface AuditLog {
    id: number;
    userEmail: string;
    action: string;
    timestamp: string;
    details: string;
}

const getAllAuditLogs = async (page: number = 0, size: number = 100): Promise<AuditLog[]> => {
    // Note: The backend returns a Page<AuditLogResponse>. We extract .content
    const response = await api.get(`${API_URL}?page=${page}&size=${size}`);
    const data = response.data.data;
    if (data && data.content) {
        return data.content;
    }
    return data || [];
};

const getAuditLogsByUser = async (userId: number): Promise<AuditLog[]> => {
    const response = await api.get(`${API_URL}/user/${userId}`);
    return response.data.data;
};

export const AuditLogService = {
    getAllAuditLogs,
    getAuditLogsByUser,
};